src/main/java/com/softserve/repository/impl/LessonRepositoryImpl.java

```diff
@@ -30,20 +30,36 @@
             + "set grouped = true "
             + "where id = :id";
 
+    private static final String UPDATE_GROUPED_TEACHER_OR_SUBJECT
+            = "update Lesson "
+            + "set subject.id = :subjectId, "
+            + " hours = :hours, "
+            + " teacher.id = :teacherId, "
+            + " lessonType = :lessonType, "
+            + " subjectForSite = :subjectForSite, "
+            + " linkToMeeting = :linkToMeeting "
+            + "where grouped = true "
+            + "and subject.id = :initialSubjectId "
+            + "and hours = :initialHours "
+            + "and teacher.id = :initialTeacherId "
+            + "and semester.id = :initialSemesterId "
+            + "and lessonType = :initialLessonType";
 
     private static final String UPDATE_GROUPED
             = "update Lesson "
             + "set subject.id = :subjectId, "
             + " hours = :hours, "
             + " teacher.id = :teacherId, "
             + " lessonType = :lessonType, "
-            + " subjectForSite = :subjectForSite "
+            + " subjectForSite = :subjectForSite, "
+            + " linkToMeeting = :linkToMeeting "
             + "where grouped = true "
             + "and subject.id = :initialSubjectId "
             + "and hours = :initialHours "
             + "and teacher.id = :initialTeacherId "
             + "and semester.id = :initialSemesterId "
-            + "and lessonType = :initialLessonType";
+            + "and lessonType = :initialLessonType "
+            + "and subjectForSite = :initialSubjectForSite";
```

Comment:
The only dissimilarity between these 2 queries is this line. Why not keep only one query?

Reply:
Because we need two different behaviors. We can create many grouped lessons with the same pair(Subject & Teacher). They are grouping about SubjectForSite parameter. When we update Subject or Teacher - we need to change all the grouped lessons with pair(Subject & Teacher). When we change the rest of parameters (any parameters without Subject or Teacher) we need to change only one lesson, without change another grouped lessons.